### ISSUE TITLE
NumPy arrays may be non-contiguous

### ISSUE THREAD

**Issue by LourensVeen** [multiscale/muscle3#74]
NumPy arrays may be non-contiguous, e.g. if you have a complex array `c` and take `c.real`. That seems to produce a view on the original data, for which `flags.c_continuous` is false, and `flags.f_continuous` is false too. And so you get this:
```
  File "libmuscle/mcp/message.py", line 145, in _data_encoder
    return _encode_grid(Grid(obj))
  File "libmuscle/mcp/message.py", line 100, in _encode_grid
    'order': order,
UnboundLocalError: local variable 'order' referenced before assignment
```
because the bit at line 74 assumes that one of these flags is always true.

The easiest short-term solution here is probably to have a fallback that makes a copy of the array (using `np.copy()`) if it's not contiguous. Should check though if we can easily avoid the extra copy and do that instead if possible.

**Comment by LourensVeen** [comment]
Fixed in 95cc9ff14f1724750234b57a3e4818f56b908a19

**Comment by LourensVeen** [comment]
Fix released with 0.4.0.
